services/models/workload.go

```diff
@@ -145,9 +145,9 @@ func (workload *WorkloadListItem) Parse(d v1beta1.Deployment) {
 	workload.Name = d.Name
 	workload.Type = DEPLOYMENT
 
-	/** Check the labels app and version required by Istio*/
-	_, workload.AppLabel = d.Labels["app"]
-	_, workload.VersionLabel = d.Labels["version"]
+	/** Check the labels app and version required by Istio in tempalte Pods*/
+	_, workload.AppLabel = d.Spec.Template.Labels["app"]
```

Comment:
As Spec.Template could not exist, it needs some protection.

Reply:
That is why has

```go
 _, workload.AppLabel = d.Spec.Template.Labels["app"]
```

Reply:
On the other hand Istio create the object Tempalte with empty values.

Template: v1.PodTemplateSpec{
						ObjectMeta: meta_v1.ObjectMeta{
							Labels: map[string]string{},
						},
					},

So the value will be "" for the first parameter **_** and for second **workload.AppLabel** if the key exist or not